_rules/error-field-is-not-identified-using-color-only-i3z4v8.md

```diff
@@ -0,0 +1,166 @@
+description: |
+  This rule checks that error fields can be identified not only by color differences but through another visual means.
+accessibility_requirements:
+  wcag21:1.4.1: # Use of Color (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  -  DOM Tree
+  -  CSS Styling
+acknowledgments:
+  authors:
+    -  Giacomo Petri
+---
+
+## Description
+
+This rule checks that errors are not identified only through color differences.
+
+## Applicability
+
+This rule applies to any [visible][] [input error][] together with:
+- its [programmatic label]; and
+- the closest error message that refers to it.
+
+## Expectation
+
+Each test target, together with its visual context, identify the presence of errors not only through color differences.
+
+A difference in hue resulting in a contrast ratio of 3:1 or greater is not considered as difference in color only.
+
+## Assumptions
+
+There are currently no assumptions
+
+## Accessibility Support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- [Understanding Success Criterion 1.4.1: Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
+- [F81 - Failure of Success Criterion 1.4.1 due to identifying required or error fields using color differences only](https://www.w3.org/WAI/WCAG21/Techniques/failures/F81.html)
+
+It is possible for an error field to be distinguishable from adjacent fields but still not have a meaningful error message. In that case, it would pass Success Criterion 1.4.1: Use of Color but still fail [Success Criterion 3.3.1: Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html) and, if applicable, [Scuccess Criterion 3.3.3: Error Suggestion](https://www.w3.org/WAI/WCAG21/Understanding/error-suggestion.html).
+
+While a difference in color resulting in a contrast ratio of 3:1 or greater is enough, it is a better practice to also use additional visual cues.
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The "first name" `input` value is empty. Its error message (the `p` element) conveys the presence of an error through its text.
+
+```html
+<label for="first_name">First Name</label>
+<input type="text" id="first_name" autocomplete="given-name" required style="border:1px solid #dd0000" aria-invalid="true" aria-errormessage="first_name_error_msg">
+<p id="first_name_error_msg" class="error_msg" style="color:#dd0000">This field is empty. Enter your first name.</p>
+<label for="last_name" style="color:#333333">Last Name</label>
+<input type="text" id="last_name" autocomplete="family-name" required style="border:1px solid #b0b0b0" value="Doe">
+```
+
+#### Passed Example 2
+
+The "phone number" `input` value doesn't match the `type` value. Its visual error indicator (the first couple of `label` and `input` elements) conveys the presence of an error through its CSS properties:
```

Comment:
This description uses "visual error indicator", which doesn't match the "error message" used in the Applicability. What does "visual error indicator" refers to?
(same on other descriptions)